### ISSUE TITLE
Add composer.json file to this

### ISSUE THREAD

**Issue by chewett** [c3js/c3#1627]
To bring c3 in line with d3 and to allow those with php projects to be able to more easily include c3 in PHP projects.

I can make a pull request for this.

Example of such composer.json https://github.com/mbostock/d3/blob/master/composer.json

**Comment by aendrew** [comment]
Wat. When did D3 start support Composer? That's weird.

@chewett Okay, cool, doesn't seem like it adds any extra maintenance. PR away!

**Comment by dfranco** [comment]
HI,
I would be really interested to see this improvement in C3js.
Can I do anything to help ?

**Comment by aendrew** [comment]
I'm closing this because D3 doesn't include a composer.json anymore and why would you anyway?

@dfranco Please explain why exactly this is useful, and why there is any benefit to managing JS using Composer, and I'll reconsider.

**Comment by dfranco** [comment]
Hi,
IMHO, the problems that Composer solves are
- you have a PHP app project (as I do) and you've a lot of libraries
- some of those libraries depends on other libraries (with specific version requirements sometime)
- you can easily add composer.json and composer.lock to git, SVN, etc. and keep an eye on it within your preferred version control system.

With composer, it's much much more easy.
- you define in your project which library your project depends on (you can also enforce a specific version)
- you don't need to worry about maintaining those library, composer install the latest stable version every time you run the $ composer update

Let's take C3js for example, it depends on D3js and doesn't work if you take latest stable version of D3js.
So for my project, I'v tried to setup a composer.json and it works smoothly.
The only things is missing is to have an "official" package on packagist.org.
P.S: there's one package but it's not maintained anymore :(

Hope this will convince you

Regards

**Comment by chewett** [comment]
I was originally going to add the composer file pull request as originally requested but when I went to do it I found d3 had removed the composer file that I added for them in their repo initially.

I haven't had time to reach out to see why it had been removed but it looks like they have squashed a lot of commits during a big refactor and move around github.

As said above many php based projects would use composer to manage their dependencies and don't run bower or similar. Therefore choosing to open ease of access for this project to php composer projects would increase its use. Personally I have moved to nvd3 as one of my more preferred d3 libraries so I won't be driving the change here.

Before c3 has a composer file the root cause probably needs to be addressed first. I welcome others to do this by approaching the people of the d3 repo.
